veps/sig-compute/10-dra-devices/vep.md

```diff
@@ -0,0 +1,760 @@
+	// +optional
+	HostDeviceStatuses []DeviceStatusInfo `json:"hostDeviceStatuses,omitempty"`
+}
+
+type DeviceStatusInfo struct {
+	// Name of the device as specified in spec.domain.devices.gpus.name or spec.domain.devices.hostDevices.name
+	Name string `json:"name"`
+	// DeviceResourceClaimStatus reflects the DRA related information for the device
+	DeviceResourceClaimStatus *DeviceResourceClaimStatus `json:"deviceResourceClaimStatus,omitempty"`
+}
+```
+
+Steps to make live migration work:
+1. DRA Status controller will include looking at target pods updating the above fields with device information
+2. Migration Controller currently, moves the migration from Scheduling to Scheduled when the target pod is running. At
+   this point the dra controller should have update the target pod information. Migration Controller should assert that
+   DRA status it sees is from the target pod by comparing the target pod name with podName in the status. The changes
+   should go here: https://github.com/kubevirt/kubevirt/blob/ada65cb7d99033baa3c096820027d08532e25c1e/pkg/virt-controller/watch/migration/migration.go#L624
+3. Once the above is asserted, the migration will continue to flow in the same way as it does today. When target
+   virt-handler will call SyncVMI on the virt-launcher, the virt-launcher will use the same code to convert the
+   deviceStatus into domxml.
+
+
+## Scalability
+
+1. virt-controller will have an additional control loop, however this will use the same shared informers of virt-controller
+   resulting in 0 additional watch calls
+1. virt-controller watch events will be properly filtered such that only changes to relevant status sections of pod and
+   vmi will trigger reconcile loop. Additional metrics for the workqueue of this control loop will be exposed
+1. At max virt-controller should generate 2 additional patch calls to vmi status in a happy path
+
+## Update/Rollback Compatibility
+
+- Since the feature will be behind a feature gate updates will not be impacted
+- Since the feature will be behind a feature gate, rollback will continue to work as long as feature gate is disabled
+- If the feature is enabled, the VMIs that use DRA devices will have to be cleaned up before attempting rollback.
+
+## Functional Testing Approach
+
+<!--
+An overview on the approaches used to functional test this design)
+-->
+- Unit tests, new code added will have optimum unit test coverage
+- An e2e test will be added that checks for the following:
+  - API fields are only populated if feature gate is enabled
+  - Once a VMI is created with DRA devices, the right spec values are set by virt-controller in virt-launcher pod
+  - Once the resourceclaims are created and allocated, the correct status values are set
+  - deletion of VMI will lead to release of resources
+  - If GPU clusters are available in CI then e2e tests will be real with a real driver installed
+  - If GPU clusters are not available the e2e tests will be mocked
```

Comment:
I think we need to make sure CI will be able to run at least one un-mocked test

Reply:
I answered this is other comment below, in alpha we can have mocked e2e test

Once a driver is available post alpha we could have an actual e2e test if it is acceptable